M2/Macaulay2/d/actors2.dd

```diff
@@ -142,8 +142,13 @@ pairs(e:Expr):Expr := (
 	    while i < length(o) do (
 	      provide Expr(Sequence(toExpr(i),o.i));
 	      i = i+1;)))
-     is o:List do pairs(Expr(o.v))   -- # typical value: pairs, BasicList, List
-     else WrongArg("a hash table, a sequence, a list, or a raw polynomial"));
+     -- # typical value: pairs, BasicList, List
+     is o:List do (
+	 r := pairs(Expr(o.v));
+	 when r
+	 is s:Sequence do list(o.Class, s, o.Mutable)
+	 else buildErrorPacket("internal error; expected a sequence"))
```

Comment:
when would `pairs(Expr(o.v))` with `o:List` fail?

Reply:
It shouldn't -- we should always get a sequence here. I don't expect we'll ever raise this error.

Reply:
Hmm, unless `pairs` raises an error... I'll look into this more.